GRADLE_PLUGIN.md

```diff
@@ -1,10 +1,268 @@
+        // Native target configuration.
+    }
+}
+```
+
+</div>
+
+Each target includes two __compilations__: `main` and `test` compiling product and test sources respectively. A compilation is an abstraction
+over a compiler invocation and described at the [corresponding page](https://kotlinlang.org/docs/reference/building-mpp-with-gradle.html#configuring-compilations).
+
+### Managing sources
+
+With the `kotlin-platform-native` plugin source sets are used to separate test and product sources. Also you can specify different sources
+for different platforms in the same source set:
+
+<div class="sample" markdown="1" theme="idea" mode="groovy">
+
+```groovy
+sourceSets {
+    // Adding target-independent sources.
+    main.kotlin.srcDirs += 'src/main/mySources'
+    
+    // Adding Linux-specific code.
+    main.target('linux_x64').srcDirs += 'src/main/linux'
+}
+```
+
+</div> 
+
+With the `kotlin-multiplatform` plugin __source__ __sets__ are also used to group sources but source files for different platforms are located in different source sets.
+For each declared target two source sets are created: `<target-name>Main` and `<target-name>Test` containing product and test soruces for this platform. Common for all
+platforms sources are located in `commonMain` and `commonTest` source sets created by default. More information about source sets can be found
+[here](https://kotlinlang.org/docs/reference/building-mpp-with-gradle.html#configuring-source-sets).
+ 
+<div class="sample" markdown="1" theme="idea" mode="groovy">
+
+```groovy
+kotlin {
+    sourceSets {
+        // Adding target-independent sources.
+        commonMain.kotlin.srcDirs += 'src/main/mySources'
+        
+        // Adding Linux-specific code.
+        linuxX64Main.kotlin.srcDirs += 'src/main/linux'
+    }
+}
+
+```
+
+</div> 
+
+### Managing dependencies
+
+With the `kotlin-platform-native` plugin dependencies are configured in a traditional for Gradle way by grouping them into configurations
+using the project `dependencies` block:
+
+<div class="sample" markdown="1" theme="idea" mode="groovy">
+
+```groovy
+dependencies {
+    implementation 'org.sample.test:mylibrary:1.0'
+    testImplementation 'org.sample.test:testlibrary:1.0'
+}
+```
+
+</div>
+
+The `kotlin-multiplatform` plugin also uses configurations under the hood but it also provides a `dependencies` block for each source set
+allowing configuring dependencies of this sources set:
+
+<div class="sample" markdown="1" theme="idea" mode="groovy">
+
+```groovy
+kotlin.sourceSets {
+    commonMain {
+        dependencies {
+            implementation 'org.sample.test:mylibrary:1.0'
+        }
+    }
+    
+    commonTest {
+        dependencies {
+            implementation 'org.sample.test:testlibrary:1.0'
+        }
+    }
+}
+```
+
+</div>
+
+Note that a dependency declared for `commonMain` or `commonTest` source set must be published using the `kotlin-multiplatform` plugin.
```

Comment:
> a dependency <...> must be published <...>
To me, the phrasing is a bit confusing as it may imply that the dependency declaration in this project somehow limits publication options for this project. Consider rephrasing to something like this:
> a module referenced by a dependency <...> must be published